client/src/main/java/io/pravega/client/security/auth/DelegationTokenProviderFactory.java

```diff
@@ -0,0 +1,74 @@
+import io.pravega.common.Exceptions;
+
+/**
+ * Creates a {@link DelegationTokenProvider} instance.
+ */
+public class DelegationTokenProviderFactory {
+
+    /**
+     * Creates a {@link DelegationTokenProvider} instance with empty token.
+     *
+     * @return a new {@link DelegationTokenProvider} instance
+     */
+    public static DelegationTokenProvider createWithEmptyToken() {
+        return new DelegationTokenProviderImpl("");
+    }
+
+    /**
+     * Creates a {@link DelegationTokenProvider} instance with null delegation token.
+     *
+     * @param controller the {@link Controller} client used for obtaining a delegation token from the Controller
+     * @param scopeName the name of the scope tied to the segment, for which a delegation token is to be obtained
+     * @param streamName the name of the stream tied to the segment, for which a delegation token is to be obtained
+     * @return a new {@link DelegationTokenProvider} instance
+     * @throws NullPointerException if {@code controller}, {@code scopeName} or {@code streamName} is null
+     * @throws IllegalArgumentException if {@code scopeName} or {@code streamName} is empty
+     */
+    public static DelegationTokenProvider create(Controller controller, String scopeName, String streamName) {
+        Preconditions.checkNotNull(controller, "Argument controller is null");
+        Exceptions.checkNotNullOrEmpty(scopeName, "scopeName");
+        Exceptions.checkNotNullOrEmpty(streamName, "streamName");
+        return new DelegationTokenProviderImpl(controller, scopeName, streamName);
+    }
+
+    /**
+     * Creates a {@link DelegationTokenProvider} instance with null delegation token.
+     *
+     * @param controller the {@link Controller} client used for obtaining a delegation token from the Controller
+     * @param segment the {@link Segment}, for which a delegation token is to be obtained
+     * @return a new {@link DelegationTokenProvider} instance
+     * @throws NullPointerException if {@code controller} or {@code segment} is null
+     */
+    public static DelegationTokenProvider create(Controller controller, Segment segment) {
+        return create(null, controller, segment);
+    }
+
+    /**
+     * Creates a {@link DelegationTokenProvider} instance with the specified delegation token.
+     *
+     * @param delegationToken an existing delegation token to populate the {@link DelegationTokenProvider} instance with
+     * @param controller  the {@link Controller} client used for obtaining a delegation token from the Controller
+     * @param segment the {@link Segment}, for which a delegation token is to be obtained
+     * @return a new {@link DelegationTokenProvider} instance
+     */
+    public static DelegationTokenProvider create(String delegationToken, Controller controller,
+                                                 Segment segment) {
+        Preconditions.checkNotNull(controller, "Argument controller is null");
```

Comment:
You do not need all these null checks here. The constructor of DelegationTokenProviderImpl already has nullchecks.

Reply:
Sure. I've removed these now.